src/core/reducers/data/products.ts

```diff
@@ -14,7 +14,11 @@ export default function updateProducts(state: State = [], action: Action): State
 
 // tslint:disable-next-line max-line-length
 export function updateMoreProducts(state: State, { payload }: Actions.ReceiveMoreProducts | Actions.ReceiveMorePastPurchaseProducts) {
-  if (state[0].index > payload[payload.length - 1].index) {
+  if (payload.length === 0) {
```

Comment:
Just wondering why you chose to be so explicit (rather than just `payload.length`)